tutorials/ssh-port-forwarding-set-up-load-testing-on-compute-engine/index.md

```diff
@@ -0,0 +1,313 @@
+
+Figure 5. Firewalls pose a communication challenge between the JMeter client and JMeter servers
+![firewalls](firewalls.jpeg)
+
+### SSH Port Forwarding Technique
+
+Secure Shell (SSH) has a functionality called SSH port forwarding (or SSH tunneling), where a connection is forwarded to a different port where the actual communication is made on the SSH connection. The port forwarding can be done either on the SSH client or the SSH server side. With port forwarding, you can set up ports that allow the JMeter client and server to connect to each other.
+
+Figure 6. Forwarding connections to a different port
+![forward connections](forw-connections.jpeg)
+
+Note that the examples “A” and “B” in Figure 6 refer to port numbers.
+
+To set up server side SSH port forwarding as shown in the examples above, you would execute the following command on your client machine (-L indicates local and -R indicates remote).
+
+`ssh -L A:127.0.0.1:B <username>@<server>`
+
+Similarly, the following command sets up the client side port forwarding.
+
+`ssh -R  B:127.0.0.1:A <username>@<server>`
+
+On Windows, you can configure SSH port forwarding by using PuTTY from “Connection” - “SSH” - “Tunnels” of PuTTY configuration in a similar manner.
+
+### SSH Port Forwarding Configuration for JMeter
+
+JMeter uses three connections: one from the client to the server using JMeter Connection, and two additional connections, one for each direction between the client and the server for RMI. You need, then, to configure three port forwardings to cover all connections. For the sake of an example, use the following port numbers as shown in Figure 7:
+
+- Port 24000 for the JMeter connection
+- Port 25000 for RMI from server to client
+- Port 26000 for RMI from client to server
+
+Figure 7. An example of a port forwarding configuration for three connections
+![forward config](forw-conf.png)
+
+The example command below sets up forwarding for 3 ports. Because you only need to set up SSH port forwarding and will not need to execute remote commands, you might want to add -N and -f options.
+
+```
+ssh -L 24000:127.0.0.1:24000   \
+-R 25000:127.0.0.1:25000   \
+-L 26000:127.0.0.1:26000 -N -f <username>@<server>
+```
+
+To use the proper port numbers, all that is left to do is to configure the JMeter client and server.
+
+**1. Configure the Client** 
+
+To configure the JMeter client, change the following items in bin/jmeter.properties:
+
+```
+remote_hosts=127.0.0.1:24000
+client.rmi.localport=25000
+mode=Statistical
+```
+
+Here are some configuration details:
+
+- **remote\_hosts** tells the JMeter client which JMeter server to use
```

Comment:
Similar to above, I don't think the backslash is needed anywhere in this doc since the underscore doesn't need to be escaped in this publishing environment.